### ISSUE TITLE
Loading after each page-change.

### ISSUE THREAD

**Issue by mektigh** [Volmarg/personal-management-system#39]
Everytime i change module on the site, example fro Goals to Notes i get the loading"dot" staying in front of the page.
forcing me to reload the page so that i can continue.

Seams to be after every "page load"

Any ideas?

**Comment by Volmarg** [comment]
Please check the dev tool in browser. 
Do You have any errors i `console` tab?
Which console do You use?

What response do You have in `network` tab?

**Comment by mektigh** [comment]
Well, the `console` have a warning saying.
`2app.js:94 [Violation] Added non-passive event listener to a scroll-blocking 'wheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952`
![error_pms](https://user-images.githubusercontent.com/10895655/90191808-d7f7f680-ddc1-11ea-8329-ffcca6ced7ad.png)
When a link is pressed the `network` tab does not load at all until i update the page.

Using latest Chrome Browse, though tested in Firefox with the same response (I know that you've specifically
 supports chrome but had to test)

**Comment by mektigh** [comment]
Looked deeper, and when being fast at updating pages after a click i get two more errors.
![Window_20200814_00_11_37](https://user-images.githubusercontent.com/10895655/90192263-ee528200-ddc2-11ea-8395-202be148e8ff.png)

No change in `network`, that only loads when page gets a refresh and the actual page loads.

**Comment by Volmarg** [comment]
This are just warnings - so that's not a case. Are You sure that in `network` You marked the `All` Filter?
Are You using the `app.js` from package or did You compiled it on Your own?

**Comment by mektigh** [comment]
Okay,
Well the `network` loads content when the page gets initially loaded.
But when clicking a link in the menu the page "loads" giving me the content behind the loading dot and the `network` does
not reload any content what so ever until page-refresh.

**Comment by Volmarg** [comment]
Interesting.

You've got Your instance online or just locally?
Which branch did You used?

**Comment by mektigh** [comment]
I'm running it only locally.
Tried a new instance on a different docker host and got the same result.

Took the `main` branch.

Really strange, could see if i can upload a GIF showing the flow if you'd like?

**Comment by Volmarg** [comment]
Try from latest `release`.
I'm generally using `master` on my local `prod` but maybe i didn't pushed something or got bit outdated state.

**Comment by mektigh** [comment]
i'll try that as soon as i get home.

**Comment by mektigh** [comment]
I've tried the version 1.23 now and get the same thing..
Maybe there is a problem on my network or something.
I mean, two different release, two different hosts the second one was a clean Ubuntu 20. Same result.

The loading starts after the page connects but as said i cannot get past the loader until i manually update the site.

![3ed42d07158b1308f023cbef64be7627](https://user-images.githubusercontent.com/10895655/90271212-bd725b80-de5b-11ea-9252-0a02a9eda88e.gif)

**Comment by Volmarg** [comment]
the one below `my-passwords` `200` `xhr` `29ms` what is the response content?

**Comment by mektigh** [comment]
Got two respones with this branch.
![image](https://user-images.githubusercontent.com/10895655/90271974-f3fca600-de5c-11ea-8d3f-1f92f5e43b98.png)

**Comment by Volmarg** [comment]
Click on the `my-passwords` and check the response

**Comment by Volmarg** [comment]
When You click on that line `my password` You will have response.

If it doesnt containt any private data then please send it here

![image](https://user-images.githubusercontent.com/37488679/90272918-6a4dd800-de5e-11ea-938c-c0be7c7a48fc.png)

**Comment by mektigh** [comment]
{
    "code": 200,
    "message": "",
    "template": "\n\n\n\n                    \n    \u003Cscript\u003E\n        var TWIG_GET_ATTRS   = \u0027[]\u0027;\n        var TWIG_ROUTE       = \u0027my-passwords\u0027;\n        var TWIG_REQUEST_URI = \u0027  \/my-passwords \u0027;\n    \u003C\/script\u003E\n\n    \u003C!-- ### $App Screen Content ### --\u003E\n    \u003Cmain class=\u0027main-content bgc-grey-100\u0027\u003E\n        \u003Cdiv id=\u0027mainContent\u0027\u003E\n            \u003Cdiv class=\u0022container-fluid\u0022\u003E\n                                \u003Cdiv class=\u0022row\u0022\u003E\n                    \u003Cdiv class=\u0022col-md-12\u0022\u003E\n                        \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-20 mB-20 d-flex flex-column\u0022\u003E\n                            \u003Ch4 class=\u0022c-grey-900 mB-20 align-self-center\u0022\u003EAdd\/Remove credentials\u003C\/h4\u003E\n\n                            \u003Ctable id=\u0022MyPasswords\u0022 class=\u0022table table-striped table-bordered\u0022 cellspacing=\u00220\u0022 data-table=\u0022true\u0022 width=\u0022100%\u0022\u003E\n    \u003Cthead\u003E\n    \u003Ctr\u003E\n        \u003Ctd class=\u0022d-none\u0022\u003EId\u003C\/td\u003E\n        \u003Ctd\u003ELogin\u003C\/td\u003E\n        \u003Ctd\u003EPassword\u003C\/td\u003E\n        \u003Ctd\u003EUrl\u003C\/td\u003E\n        \u003Ctd\u003EDescription\u003C\/td\u003E\n        \u003Ctd\u003EGroup\u003C\/td\u003E\n        \u003Ctd\u003EActions\u003C\/td\u003E\n    \u003C\/tr\u003E\n    \u003C\/thead\u003E\n    \u003Ctbody\u003E\n        \u003C\/tbody\u003E\n\u003C\/table\u003E\n\n                            \u003Cdiv class=\u0022col-md-6 align-self-center add-record-form\u0022\u003E\n                                \u003Cform name=\u0022my_passwords\u0022 method=\u0022post\u0022 data-entity=\u0022MyPasswords\u0022\u003E\n                                \u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2 required\u0022 for=\u0022my_passwords_login\u0022\u003ELogin\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cinput type=\u0022text\u0022 id=\u0022my_passwords_login\u0022 name=\u0022my_passwords[login]\u0022 required=\u0022required\u0022 class=\u0022form-control\u0022 \/\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2 required\u0022 for=\u0022my_passwords_password\u0022\u003EPassword\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cinput type=\u0022text\u0022 id=\u0022my_passwords_password\u0022 name=\u0022my_passwords[password]\u0022 required=\u0022required\u0022 class=\u0022form-control\u0022 \/\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2\u0022 for=\u0022my_passwords_url\u0022\u003EUrl\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cinput type=\u0022text\u0022 id=\u0022my_passwords_url\u0022 name=\u0022my_passwords[url]\u0022 class=\u0022form-control\u0022 \/\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2 required\u0022 for=\u0022my_passwords_description\u0022\u003EDescription\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cinput type=\u0022text\u0022 id=\u0022my_passwords_description\u0022 name=\u0022my_passwords[description]\u0022 required=\u0022required\u0022 class=\u0022form-control\u0022 \/\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2 required\u0022 for=\u0022my_passwords_group\u0022\u003EGroup\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cselect id=\u0022my_passwords_group\u0022 name=\u0022my_passwords[group]\u0022 class=\u0022form-control\u0022\u003E\u003C\/select\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Cdiv class=\u0022col-sm-2\u0022\u003E\u003C\/div\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cbutton type=\u0022submit\u0022 id=\u0022my_passwords_submit\u0022 name=\u0022my_passwords[submit]\u0022 class=\u0022btn-primary btn\u0022\u003ESubmit\u003C\/button\u003E\u003C\/div\u003E\u003C\/div\u003E\u003Cinput type=\u0022hidden\u0022 id=\u0022my_passwords__token\u0022 name=\u0022my_passwords[_token]\u0022 value=\u0022nvwRDwXBSpMJJXu2n8DdjNFPi0uEaniM5WY3A9C17oc\u0022 \/\u003E\u003C\/form\u003E\n                            \u003C\/div\u003E\n                        \u003C\/div\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n            \u003C\/div\u003E\n        \u003C\/div\u003E\n    \u003C\/main\u003E\n\n\u003Csection class=\u0022global-messages\u0022\u003E\n\n    \n    \n        \n    \n        \n    \n\u003C\/section\u003E\n\n\u003Cscript type=\u0022module\u0022\u003E\n\n    \/\/ prevent overwriting existing variable (js crashes)\n    if( \u0022object\u0022 === typeof window.messages )\n    {\n        delete window.messages;\n    }\n\n    window.messages = document.querySelectorAll(\u0027.global-messages .alert\u0027);\n\n    \/\/ show message and remove it\n    window.messages.forEach((element) =\u003E {\n        let type    = element.getAttribute(\u0027data-type\u0027);\n        let message = element.innerHTML;\n\n        window.legacy_notify(message, type);\n        element.parentNode.removeChild(element);\n    });\n\n\u003C\/script\u003E",
    "reload_page": false,
    "reload_message": "",
    "success": true,
    "form_template": "",
    "validated_form_prefix": "",
    "invalid_form_fields": []
}

**Comment by mektigh** [comment]
And that type of response i get on every menu-item going anywhere.
When updated/reloaded the page works flawless until updated/another button pressed.

**Comment by Volmarg** [comment]
so You can navigate between pages and the problem begins when You update something?

**Comment by mektigh** [comment]
Logging in, getting to dashboard works great.
Clicking on ex. Passwords, the problem begins.
Refreshing the page, the password page works as intended
clicking another item on the menu, the page loads and the loadingscreen gets stuck over the page.
Reloading makes the page work as intended again.

So no matter where i am, if i want to change view to a different module i have to klick the item, and when loader starts, reload the page to be able to use the page.

**Comment by Volmarg** [comment]
I will check that response later and the steps You've described. Will then let know what I've found out.

**Comment by mektigh** [comment]
Nice,
Theres no hurry.

From the page Passwords clicking on Achievements: Response:
```
{
    "code": 200,
    "message": "",
    "template": "\n\n\n\n                    \n    \u003Cscript\u003E\n        var TWIG_GET_ATTRS   = \u0027[]\u0027;\n        var TWIG_ROUTE       = \u0027achievement\u0027;\n        var TWIG_REQUEST_URI = \u0027  \/achievement \u0027;\n    \u003C\/script\u003E\n\n\u003C!-- ### $App Screen Content ### --\u003E\n\u003Cmain class=\u0027main-content bgc-grey-100\u0027\u003E\n    \u003Cdiv id=\u0027mainContent\u0027\u003E\n        \u003Cdiv class=\u0022container-fluid\u0022\u003E\n                        \u003Cdiv class=\u0022row\u0022\u003E\n                \u003Cdiv class=\u0022col-md-12\u0022\u003E\n                    \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-10 mB-20 d-flex justify-content-center\u0022\u003E\n                        \u003Ch4 class=\u0022c-grey-900 mB-20\u0022\u003EAchievements\u003C\/h4\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n            \u003C\/div\u003E\n                        \u003Cdiv class=\u0022row\u0022\u003E\n                                \u003Cdiv class=\u0022col-md-3\u0022\u003E\n                    \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-10 mB-10\u0022\u003E\n                        \u003Ch5 class=\u0022c-grey-900 mB-20 text-center\u0022\u003ESimple\u003C\/h5\u003E\n                        \u003Cdiv\u003E\n    \u003Ctable id=\u0022Achievements\u0022 class=\u0022table\u0022 cellspacing=\u00220\u0022\n           data-table=\u0022false\u0022 width=\u0022100%\u0022\u003E\n        \u003Ctbody\u003E\n                \u003C\/tbody\u003E\n    \u003C\/table\u003E\n\u003C\/div\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n                                \u003Cdiv class=\u0022col-md-3\u0022\u003E\n                    \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-10 mB-10\u0022\u003E\n                        \u003Ch5 class=\u0022c-grey-900 mB-20 text-center\u0022\u003EMedium\u003C\/h5\u003E\n                        \u003Cdiv\u003E\n    \u003Ctable id=\u0022Achievements\u0022 class=\u0022table\u0022 cellspacing=\u00220\u0022\n           data-table=\u0022false\u0022 width=\u0022100%\u0022\u003E\n        \u003Ctbody\u003E\n                \u003C\/tbody\u003E\n    \u003C\/table\u003E\n\u003C\/div\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n                                \u003Cdiv class=\u0022col-md-3\u0022\u003E\n                    \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-10 mB-10\u0022\u003E\n                        \u003Ch5 class=\u0022c-grey-900 mB-20 text-center\u0022\u003EHard\u003C\/h5\u003E\n                        \u003Cdiv\u003E\n    \u003Ctable id=\u0022Achievements\u0022 class=\u0022table\u0022 cellspacing=\u00220\u0022\n           data-table=\u0022false\u0022 width=\u0022100%\u0022\u003E\n        \u003Ctbody\u003E\n                \u003C\/tbody\u003E\n    \u003C\/table\u003E\n\u003C\/div\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n                                \u003Cdiv class=\u0022col-md-3\u0022\u003E\n                    \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-10 mB-10\u0022\u003E\n                        \u003Ch5 class=\u0022c-grey-900 mB-20 text-center\u0022\u003EHardcore\u003C\/h5\u003E\n                        \u003Cdiv\u003E\n    \u003Ctable id=\u0022Achievements\u0022 class=\u0022table\u0022 cellspacing=\u00220\u0022\n           data-table=\u0022false\u0022 width=\u0022100%\u0022\u003E\n        \u003Ctbody\u003E\n                \u003C\/tbody\u003E\n    \u003C\/table\u003E\n\u003C\/div\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n                            \u003C\/div\u003E\n                        \u003Cdiv class=\u0022row\u0022\u003E\n                \u003Cdiv class=\u0022col-md-12\u0022\u003E\n                    \u003Cdiv class=\u0022bgc-white bd bdrs-3 p-20 mB-20 d-flex flex-column\u0022\u003E\n                        \u003Ch4 class=\u0022c-grey-900 mB-20 align-self-center\u0022\u003EAdd achievements\u003C\/h4\u003E\n                        \u003Cdiv class=\u0022col-md-6 align-self-center add-record-form\u0022\u003E\n                            \u003Cform name=\u0022achievement\u0022 method=\u0022post\u0022 data-entity=\u0022Achievements\u0022\u003E\n                            \u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2 required\u0022 for=\u0022achievement_Name\u0022\u003EName\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cinput type=\u0022text\u0022 id=\u0022achievement_Name\u0022 name=\u0022achievement[Name]\u0022 required=\u0022required\u0022 maxlength=\u0022255\u0022 class=\u0022form-control\u0022 \/\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2\u0022 for=\u0022achievement_Description\u0022\u003EDescription\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cinput type=\u0022text\u0022 id=\u0022achievement_Description\u0022 name=\u0022achievement[Description]\u0022 maxlength=\u0022255\u0022 class=\u0022form-control\u0022 \/\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Clabel class=\u0022col-form-label col-sm-2 required\u0022 for=\u0022achievement_Type\u0022\u003EType\u003C\/label\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cselect id=\u0022achievement_Type\u0022 name=\u0022achievement[Type]\u0022 class=\u0022form-control\u0022\u003E\u003Coption value=\u0022simple\u0022\u003ESimple\u003C\/option\u003E\u003Coption value=\u0022medium\u0022\u003EMedium\u003C\/option\u003E\u003Coption value=\u0022hard\u0022\u003EHard\u003C\/option\u003E\u003Coption value=\u0022hardcore\u0022\u003EHardcore\u003C\/option\u003E\u003C\/select\u003E\u003C\/div\u003E\n    \u003C\/div\u003E\u003Cdiv class=\u0022form-group row\u0022\u003E\u003Cdiv class=\u0022col-sm-2\u0022\u003E\u003C\/div\u003E\u003Cdiv class=\u0022col-sm-10\u0022\u003E\u003Cbutton type=\u0022submit\u0022 id=\u0022achievement_submit\u0022 name=\u0022achievement[submit]\u0022 class=\u0022btn-primary btn\u0022\u003ESubmit\u003C\/button\u003E\u003C\/div\u003E\u003C\/div\u003E\u003Cinput type=\u0022hidden\u0022 id=\u0022achievement__token\u0022 name=\u0022achievement[_token]\u0022 value=\u0022vZCuhpkKmfHB4ya0rEjxkWwotOmBDmncMyv9zYH-l2s\u0022 \/\u003E\u003C\/form\u003E\n                        \u003C\/div\u003E\n                    \u003C\/div\u003E\n                \u003C\/div\u003E\n            \u003C\/div\u003E\n        \u003C\/div\u003E\n    \u003C\/div\u003E\n\u003C\/main\u003E\n\n\u003Csection class=\u0022global-messages\u0022\u003E\n\n    \n    \n        \n    \n        \n    \n\u003C\/section\u003E\n\n\u003Cscript type=\u0022module\u0022\u003E\n\n    \/\/ prevent overwriting existing variable (js crashes)\n    if( \u0022object\u0022 === typeof window.messages )\n    {\n        delete window.messages;\n    }\n\n    window.messages = document.querySelectorAll(\u0027.global-messages .alert\u0027);\n\n    \/\/ show message and remove it\n    window.messages.forEach((element) =\u003E {\n        let type    = element.getAttribute(\u0027data-type\u0027);\n        let message = element.innerHTML;\n\n        window.legacy_notify(message, type);\n        element.parentNode.removeChild(element);\n    });\n\n\u003C\/script\u003E",
    "reload_page": false,
    "reload_message": "",
    "success": true,
    "form_template": "",
    "validated_form_prefix": "",
    "invalid_form_fields": []
}
```
Working flawless after browser refresh.

Note: Inactivated all addons and such on the browser, no change.

**Comment by Volmarg** [comment]
The response is ok.

I've got other idea since i also had such problem twice, it might be that for other ppl this still continues to exist.

First of all. Did You recompiled the js or do You just use what's in the branch/package? There was already someone who compiled it with wrong node version so he had the same issue.

Check out this steps:
1. Login
 - Dashboard should open,
 - take a look on the page content,
2. Open  `achievements`
- does the page content changes?

**Comment by mektigh** [comment]
Thanks for the fast replys mate.

Well, I did not recompile the js but used the one in the package.
Though i have not tried another version of the node, will check that in a moment.

1. Login `ok`
 - The Page loads as it should.
2. Click `Achievements` `ok`
 - The page loads
 `error`
 - The loading screen just loops on top of the page.

Tried with Node 10.22.0, 9.11.2 and lts. 
Same result, the bug presists.

**Comment by Volmarg** [comment]
Can You check if this also happens with the `images` module?

But make a test:
1. Go to images module,
2. Open dev console in browser
3. Call this `ui.widgets.loader.hideLoader();`
4. Upload some pictures
5. Then open the project in new window and try to enter the images module again
- does it keep spinning?

**Comment by mektigh** [comment]
Okay,
So, i went to the Images tab, actually got no problem loading the thing, uploaded pictures, they uploaded without problem.
Called `ui.widgets.loader.hideLoader();`
Went opening a new tab, opened the page and clicked images but the spinning showed as usual.
Though ofcourse not on the `tab `with `ui.widgets.loader.hideLoader();`

**Comment by Volmarg** [comment]
Is that the same case? http://185.204.216.136:888/ 

It's seems that something like that happens only sometimes here.

But if that's the case then I can pretty much search further with recreated issue.

**Comment by Volmarg** [comment]
L:admin / P: admin

**Comment by mektigh** [comment]
Yeah, that page runs exactly like my hosted one.
Same bug 👍

**Comment by Volmarg** [comment]
Ok, this will need to wait for next weekend. But so far if You want to work with the project You can try to remove in Your loca instance lines `102-104` : https://github.com/Volmarg/personal-management-system/blob/main/templates/base.html.twig#L103

- might just work and permanently disable the  Loader

**Comment by mektigh** [comment]
Oi,
Sorry for the delayed response.
I removed the lines `102-104` and at first it didnt change anything.
ran the following and started again:
`sudo bin/console cache:clear`
`sudo bin/console cache:warmup`
`sudo bin/console doctrine:schema:update --env=dev --force`

`bin/console --env=dev server:run 0.0.0.0:8005`

The problem `loader` is now gone 👍

**Comment by otisdog8** [comment]
I host this through nginx and php-fpm (rather than the default dev server), and this solution also worked for me.

**Comment by Volmarg** [comment]
Interesting. But where is the difference of how You cleared, created db. Structure?

**Comment by Volmarg** [comment]
So I've updated the: http://185.204.216.136:888/
With the currently released 1.3 (TS update) and I can't recreate that bug, but might be just me.

**Comment by Volmarg** [comment]
Fixed: https://github.com/Volmarg/personal-management-system/commit/79be71303c524dba043a270fa4f8a8facddd6022
